x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/threat_marker/scan.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import type { Client } from '@elastic/elasticsearch';
+import async from 'async';
+
+export interface Threat {
+  'threat.indicator.type': string;
+  'threat.indicator.url.full': string;
+  'threat.indicator.file.hash.md5': string;
+  'threat.indicator.file.hash.sha1': string;
+}
+
+import type {
+  OpenPointInTimeResponse,
+  SortResults,
+  QueryDslQueryContainer,
+  SearchHit,
+} from '@elastic/elasticsearch/lib/api/types';
+
+/**
+ * Returns filter clause with 'match' conditions for relevant fields only
+ * @param threat
+ * @returns
+ */
+export const shouldClauseForThreat = (threat: Threat) => {
+  switch (threat['threat.indicator.type']) {
+    case 'url': {
+      return [
+        {
+          match: {
+            'url.full': threat['threat.indicator.url.full'],
+          },
+        },
+      ];
+    }
+
+    case 'file': {
+      return [
+        {
+          match: {
+            'file.hash.md5': threat['threat.indicator.file.hash.md5'],
+            'file.hash.sha1': threat['threat.indicator.file.hash.sha1'],
+          },
+        },
+      ];
+    }
+  }
+};
+
+export const THREAT_DETECTION_INDICATOR_FIELD = 'threat.detection.indicator' as const;
```

Comment:
there is a field `threat.enrichment.indicator` https://www.elastic.co/guide/en/ecs/current/ecs-threat.html#field-threat-enrichments that exists in the ECS. Though `threat.enrichment` is a list of objects. Did you consider using it or is this field being a list of objects the reason making it not suitable for this use case?

Reply:
we can use that for sure, just wanted to store it under some name really.